apps/blas/rot/rot-alt.fil

```diff
@@ -0,0 +1,131 @@
+import "apps/blas/util.fil";
+import "primitives/signed.fil";
+
+// Applies a rotation to vectors x, y
+// W: element width
+// N: vector length
+// M: multiplier amount
+// A: adder amount
+// Tiles scalar-vector products based on multiplier count,
+// then does additions as multiplies finish
+comp Rot[W, N, M, A]<'G:II>(
+  go: interface['G],
+  c: ['G, 'G+1] W,
+  s: ['G, 'G+1] W,
+  x[N]: ['G, 'G+1] W,
+  y[N]: ['G, 'G+1] W,
+) -> (
+  out_1[N]: ['G+L, 'G+L+1] W,
+  out_2[N]: ['G+L, 'G+L+1] W
+) with {
+  some L where L > 0;
+  some II where II > 0;
+} where W > 0,
+        N > 0,
+        M > 0,
+        A > 0,
+        M % 4 == 0,   // need to do at least 4 multiplies at once
```

Comment:
Tangential fly-by comment:  maybe we should add some syntax to allow asserts to add a reason for the assertion so we can generate better errors. For example when one of these asserts trigger, the comment could show up as the rationale for the restriction.